server/frontend/src/components/Comments/PublicationForm.vue

```diff
@@ -0,0 +1,492 @@
+          class="alert alert-danger"
+          role="alert"
+        >
+          <button
+            type="button"
+            class="close"
+            data-dismiss="alert"
+            aria-label="Close"
+            v-on:click="publishTestCaseError = null"
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+          An error occurred while attaching the testcase to the created external
+          comment: {{ publishTestCaseError }}
+        </div>
+
+        <div v-if="!bugzillaToken" class="alert alert-warning" role="alert">
+          Please define an API Token for
+          <strong>{{ provider.hostname }}</strong> in your settings to submit a
+          new comment.
+        </div>
+
+        <div>
+          <button
+            type="button"
+            class="btn btn-danger"
+            v-on:click="createExternalComment"
+            :disabled="!bugzillaToken || submitting"
+          >
+            Submit
+          </button>
+          <button
+            type="button"
+            class="btn btn-default"
+            v-on:click="goBack"
+            :disabled="submitting"
+          >
+            Cancel
+          </button>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+var Mustache = require("mustache");
+import { errorParser } from "../../helpers";
+import * as api from "../../api";
+import * as bugzillaApi from "../../bugzilla_api";
+import CrashDataSection from "../Bugs/CrashDataSection.vue";
+import TestCaseSection from "../Bugs/TestCaseSection.vue";
+
+export default {
+  components: {
+    CrashDataSection,
+    TestCaseSection,
+  },
+  props: {
+    providerId: {
+      type: Number,
+      required: true,
+    },
+    templateId: {
+      type: Number,
+      required: true,
+    },
+    entryId: {
+      type: Number,
+      required: true,
+    },
+    bugId: {
+      type: Number,
+      required: false,
+      default: null,
+    },
+  },
+  data: () => ({
+    providers: [],
+    selectedProvider: null,
+    provider: null,
+    templates: [],
+    selectedTemplate: null,
+    template: null,
+    entry: null,
+    externalBugId: "",
+    isPrivate: false,
+    submitting: false,
+    createError: null,
+    publishTestCaseError: null,
+    publishCrashDataError: null,
+    createdCommentId: null,
+    createdCommentCount: null,
+    notAttachTest: false,
+    testCaseContent: "",
+    notAttachData: false,
+    crashData: "",
+  }),
+  async mounted() {
+    this.entry = await api.retrieveCrash(this.entryId);
+    this.crashData = this.entry.rawCrashData
+      ? this.entry.rawCrashData
+      : this.entry.rawStderr;
+    if (this.bugId) this.externalBugId = this.bugId;
+
+    let data = await api.listBugProviders();
+    this.providers = data.results.filter(
+      (p) => p.classname === "BugzillaProvider"
+    );
+    this.provider = this.providers.find((p) => p.id === this.providerId);
+    this.selectedProvider = this.provider.id;
```

Comment:
Could the situation ever occur where this page is accessed yet no BugzillaProvider has been registered?

Reply:
I kept the Django view code as is, it prevents accessing the component if no provider exists yet since it will raise a 404 error before rendering:
```
if request.method == 'GET':
    if 'provider' in request.GET:
        provider = get_object_or_404(BugProvider, pk=request.GET['provider'])
    else:
        user = User.get_or_create_restricted(request.user)[0]
        provider = get_object_or_404(BugProvider, pk=user.defaultProviderId)
```